src/engine/server/server.cpp

```diff
@@ -3995,6 +4108,74 @@ const char *CServer::GetAnnouncementLine()
 	return m_vAnnouncements[m_AnnouncementLastLine].c_str();
 }
 
+struct CSubdirCallbackUserdata
+{
+	CServer *m_pServer;
+	char m_aCurrentFolder[IO_MAX_PATH_LENGTH];
+};
+
+int CServer::MaplistEntryCallback(const char *pFilename, int IsDir, int DirType, void *pUser)
+{
+	CSubdirCallbackUserdata *pUserdata = static_cast<CSubdirCallbackUserdata *>(pUser);
+	CServer *pThis = pUserdata->m_pServer;
+
+	if(pFilename[0] == '.' && (pFilename[1] == '\0' || (pFilename[1] == '.' && pFilename[2] == '\0'))) // hidden files
```

Comment:
```suggestion
	if(str_comp(pFilename, ".") == 0 || str_comp(pFilename, ".."))
```
What does the `// hidden files` do there? If I read this correctly, it just hides "." and "..". The change makes it clearer to me.